clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java

```diff
@@ -176,10 +183,37 @@ public void close()  {
         loginManagers.clear();
     }
 
-    private TransportLayer buildTransportLayer(String id, SelectionKey key, SocketChannel socketChannel) throws IOException {
+    public static <U> U logSlow(Callable<U> fn, String msg) throws IOException {
+        long before = System.nanoTime();
+        U result = null;
+        try {
+            result = fn.call();
+        } catch (Exception e) {
+            throw new IOException(e);
```

Comment:
@gardnervickers : Could we just propagate the original exception instead of wrapping in inside IOException?